### DISCUSSION TITLE
ParseErrors and Laravel Octane

### DISCUSSION THREAD

**Discussion by freshleafmedia** [spatie/laravel-ignition#155] [Q&A]
When running Laravel via Octane (either Swoole or Roadrunner v3) `ParseError` exceptions cause the server to exit and a completely empty response is returned (aka white screen). The exception is printed to the console but I expected that the exception would be caught and displayed like normal.

Repro:

1. `composer create-project laravel/laravel /tmp/laravel-repro`
2. `cd /tmp/laravel-repro`
3. `composer require laravel/octane`
4. `php artisan octane:install`
5. `php artisan octane:start --poll`
6. write some invalid code:
   ```
   <?php
    
    Illuminate\Support\Facades\Route::get('/', function () {
        nope
    });
    ```
7. `GET /`

Expected:

![20230720-1713](https://github.com/spatie/laravel-ignition/assets/10062339/8d02c6c5-0bd3-409c-9c64-634dd765fb1f)

Got:

_blank screen_

I note that the `octane:start` command prints the exception to the console and then exits:

```
$ php artisan octane:start --poll

   INFO  Server running…

  Local: http://127.0.0.1:8000 

  Press Ctrl+C to stop the server


   ParseError 

  syntax error, unexpected token "}"

  at routes/web.php:5
      1▕ <?php
      2▕ 
      3▕ Illuminate\Support\Facades\Route::get('/', function () {
      4▕     nope
  ➜   5▕ });
      6▕ 
```

**Comment by mihob** [comment]
Any news on this?
